trace/exemplar.go

```diff
@@ -0,0 +1,52 @@
+// Copyright 2018, OpenCensus Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package trace
+
+import (
+	"context"
+	"encoding/hex"
+	"fmt"
+
+	"go.opencensus.io/exemplar"
+)
+
+// Attachment keys for sampled trace span active at the time of recording.
+const (
+	ExemplarKeyTraceID      = "TraceID"
+	ExemplarKeySpanID       = "SpanID"
+	ExemplarKeyTraceOptions = "TraceOptions"
+)
+
+func init() {
+	exemplar.RegisterAttachmentExtractor(attachSpanContext)
+}
+
+func attachSpanContext(ctx context.Context, attachments map[string]string) map[string]string {
+	span := FromContext(ctx)
+	sc := span.SpanContext()
```

Comment:
This might result in a nil dereference on the first span(if SpanContext doesn't check that it is nil), but anyways we don't use it if span == nil. Please move it below after we've checked `if span == nil`